src/components/ParserComponent.tsx

```diff
@@ -143,81 +144,95 @@ export default class ParserComponent extends React.Component<ParserInterface,
       assert.notStrictEqual(data, null,
-          'sortData(): data (Array of objects) is null');
+          'dataIsValid(): data (Array of objects) is null');
       assert.notStrictEqual(data, [],
-          'sortData(): data (array of objects) is empty');
-
-      let doneTheWork = false;
-      /* loop goes through each key and saves the 1 with a date in first row */
+          'dataIsValid(): data (array of objects) is empty');
       if (data !== undefined && data.length > 0) {
-        assert.notStrictEqual(data[0], null,
-            'sortData(): data[0] is null');
-        for (const [key, value] of Object.entries(data[0])) {
-          if (!doneTheWork) {
-            const date1 = moment(String(value), this.state.formatString);
-            if (moment(date1, this.state.formatString).isValid()) {
-              doneTheWork = true;
-              const formatString = this.state.formatString;
-
-              const keyInt = `${key}_num`;
-
-              TimSort.sort(data, function(a: any, b: any) {
-                let val: any;
-                if (!a.hasOwnProperty(keyInt)) {
-                  val = moment(a[key], formatString);
-                  if (val.isValid()) {
-                    a[keyInt] = val.valueOf();
-                  } else {
-                    a[keyInt] = -1;
-                  }
-                }
-
-                if (!b.hasOwnProperty(keyInt)) {
-                  val = moment(b[key], formatString);
-                  if (val.isValid()) {
-                    b[keyInt] = val.valueOf();
-                  } else {
-                    b[keyInt] = -1;
-                  }
-                }
-                return (a[keyInt] - b[keyInt]);
-              });
-
-              this.setState(() => {
-                return {
-                  data,
-                };
-              });
-            }
-          }
-        }
-      }
-      if (doneTheWork) {
-        // state should be updated
-        assert.notStrictEqual(this.state.data, [],
-            'sortData(): this.state.data is empty (not updated)');
         return true;
       } else {
-        throw new Error('The file uploaded has no dates.');
+        throw new Error('The file uploaded has no data.');
       }
     }
 
+    /**
+     * purpose: looks for the column that has a valid date in the first row
+     * @precondition a csv has been uploaded and its data is stored in an array
+     * @param {Array} data: the array of data to be checked
+     * @return {any}: return the column name of the first row
+     * that contains the valid date of given date format
+   */
+    lookForDateKey(data: Array<object>): any {
+      assert.notStrictEqual(data[0], null,
+          'sortData(): data[0] is null');
+      for (const [key, value] of Object.entries(data[0])) {
+        const date1 = moment(String(value), this.state.formatString);
+        if (moment(date1, this.state.formatString).isValid()) {
+          return key;
+        }
```

Comment:
If no date column is found, then this doesn't return anything or throw an error.